### DISCUSSION TITLE
ReferenceError: regeneratorRuntime  is not defined

### DISCUSSION THREAD

**Discussion by Garajluc** [vercel/next.js#40537] [Help]
### Verify canary release

- [ ] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

Operating System:
      Platform: linux
      Arch: x64
      Version: #23~20.04.1-Ubuntu SMP Thu Aug 18 03:20:14 UTC 2022
    Binaries:
      Node: 18.8.0
      npm: 8.18.0
      Yarn: 1.22.19
      pnpm: N/A
    Relevant packages:
      next: 12.3.0
      eslint-config-next: 12.1.6
      react: 18.1.0
      react-dom: 18.1.0

### What browser are you using? (if relevant)

n/a

### How are you deploying your application? (if relevant)

n/a

### Describe the Bug

When upgrading to the latest version of next "12.3.0", we are getting this error: `ReferenceError: regeneratorRuntime  is not defined`. This error is introduced by upgrading the Next.js version from `12.1.6` to `12.3.0`.  `useAsyncDebounce` is used in the project by the library `react-table v7.8.0`. 

<img width="1072" alt="Screenshot 2022-09-13 at 14 46 48" src="https://user-images.githubusercontent.com/32719289/189925887-a5d62746-a32a-4c00-ac90-225c35e8aec2.png">

What I have tried to fix it: 
- I have already tried to follow some of the solutions related to the actual library `react-table` (such as: https://github.com/TanStack/table/issues/3297) but it didn't solve the issue. 

### Expected Behavior
The regeratorRuntime should be defined - the client-side error should not occur. 

### Link to reproduction

n/a

### To Reproduce

use react-table v7.8.0 in you project and use functionality of react-table which is using `useAsyncDebounce`.

**Comment by taylorhlabs** [comment]
I had this same issue so I removed the useAsyncDebounce hook and created a simple debounce function based on an example for Tanstack Table v8 here: https://codesandbox.io/s/github/tanstack/table/tree/main/examples/react/filters?from-embed=&file=/src/main.tsx:11951-12106

My search term is managed using 
```
const [searchTerm, setSearchTerm] = useState(initialValue);
```
so replace `useAsyncDebounce` with
```
useEffect(() => {
  const timeout = setTimeout(() => {
    handleSetSearchTerm(searchTerm);
  }, 500);

  return () => clearTimeout(timeout);
}, [searchTerm, handleSetSearchTerm]);
```
 and update the `onChange` function in your Input component to
```
onChange={(e) => setSearchTerm(e.target.value)}
```

Full implementation (I'm using Chakra UI):
```
export default function SearchBar({
  initialValue = "",
  numberResults,
  typeResults,
  handleSetSearchTerm
}: {
  initialValue?: string;
  numberResults: number;
  typeResults: string;
  handleSetSearchTerm: (value: string) => void;
}) {
  const [searchTerm, setSearchTerm] = useState(initialValue);

  useEffect(() => {
    const timeout = setTimeout(() => {
      handleSetSearchTerm(searchTerm);
    }, 500);

    return () => clearTimeout(timeout);
  }, [searchTerm, handleSetSearchTerm]);

  return (
    <InputGroup size="md">
      <InputLeftElement pointerEvents="none">
        <FontAwesomeIcon
          icon={faSearch}
          className="h-5 w-5 text-gray-400"
          aria-hidden="true"
        />
      </InputLeftElement>
      <Input
        pr="4.5rem"
        type="text"
        name="search_text"
        id="search_text"
        onChange={(e) => setSearchTerm(e.target.value)}
        placeholder={
          numberResults === 1
            ? `Search 1 ${typeResults}`
            : `Search ${numberResults} ${typeResults}s`
        }
        value={searchTerm}
      />
      <InputRightElement width="4.5rem">
        <Button
          title="Clear Search"
          size="sm"
          onClick={() => {
            setSearchTerm("");
            handleSetSearchTerm("");
          }}
          disabled={searchTerm ? false : true}
        >
          <FontAwesomeIcon icon={faTimes} />
        </Button>
      </InputRightElement>
    </InputGroup>
  );
}
```

**Reply by Nelio-Bila** [reply]
this work like a charm! thank you

**Reply by anmol9898** [reply]
Thanks works very well

**Comment by riuci-admin** [comment]
Same issue here, any updates on this one? other that the useState code above?
